webpack/assets/javascripts/react_app/common/hooks/useSubmitOnEnter.js

```diff
@@ -0,0 +1,24 @@
+import { useEffect } from 'react';
+
+/**
+ * A custom hook that listens for enter on a ref and runs a function on enter keydown
+ * @param  {object} ref a ref created with React's useRef or createRef that scopes the event listener
+ * @param  {function} onSubmit function to execute on enter keydown
+ */
+export const useSubmitOnEnter = (ref, onSubmit) => {
+  const listener = event => {
+    if (event.code === 'Enter' || event.code === 'NumpadEnter') onSubmit();
+  };
+
+  return useEffect(() => {
+    const refCopy = ref; // re-assigning so it's available on unmount
+    if (refCopy?.current) refCopy.current.addEventListener('keydown', listener);
```

Comment:
```suggestion
    refCopy?.current?.addEventListener?.('keydown', listener);
```

Reply:
I did have to disable an eslint rule for this:
```
  15:5  error    Expected an assignment or function call and instead saw an expression                                                    no-unused-expressions
```